Add release pipeline for this project using github actions

bbucher-sigsci opened the pull request:
https://github.com/signalsciences/sigsci/issues/23974

The action is triggered when a release is created. The workflow creates a tar.gz artifact of the bin/ and lib/ dirs, and uploads it to the release.

bbucher-sigsci committed:
force push

bbucher-sigsci committed:
do release via api

bbucher-sigsci committed:
remove uses

bbucher-sigsci committed:
testing

bbucher-sigsci committed:
remove README-SIGSCI.md

bbucher-sigsci committed:
fix json

bbucher-sigsci committed:
remove README after copying contents to README-SIGSCI.md

bbucher-sigsci committed:
test changing VERSION

bbucher-sigsci committed:
- remove rename-readme.sh
- add comments for clarity
- update VERSION to test

bbucher-sigsci committed:
test line breaks in release body text

bbucher-sigsci committed:
test line breaks in release body text

bbucher-sigsci committed:
more newline testing

bbucher-sigsci committed:
more newline testing

bbucher-sigsci committed:
grab commit messages and add them to release body

bbucher-sigsci committed:
close quote

bbucher-sigsci committed:
remove extra quotes

bbucher-sigsci committed:
change branch to master

leaf-fastly commented:
Why are we making a branch on every push to master?
Which README are we worried about conflicting with?
The main point of the issue is to provide github releases to point users to; if a .tgz/zip of the repository isn't the right format, then generating a .tgz with the correct contents can be part of the release action, but no branch should need to be created.
Let's talk.

bbucher-sigsci committed:
- rename workflow to reflect it's purpose
- run on release creation
- get upload url from context
- upload asset to release

bbucher-sigsci committed:
remove unnecessary steps

bbucher-sigsci committed:
remove VERSION

bbucher-sigsci committed:
refactor

bbucher-sigsci committed:
try published

bbucher-sigsci committed:
testing

bbucher-sigsci committed:
testing

bbucher-sigsci committed:
try other types

bbucher-sigsci committed:
testing

bbucher-sigsci committed:
Merge branch 'master' into bbucher-23974-release-pipeline

# Conflicts:
#	.github/workflows/build.yml

bbucher-sigsci committed:
replace release.yml with build.yml

bbucher-sigsci committed:
rename

bbucher-sigsci committed:
Merge branch 'master' into bbucher-23974-release-pipeline

# Conflicts:
#	.github/workflows/release.yml

bbucher-sigsci committed:
Merge branch 'master' into bbucher-23974-release-pipeline

bbucher-sigsci committed:
add release instructions

leaf-fastly reviewed (approved):
🎉
